File: aoj/volume2/n0296/Main.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int oneYenCoin = sc.nextInt();
        int fiveYenCoin = sc.nextInt() * 5;
        int tenYenCoin = sc.nextInt() * 10;
        int fiftyYenCoin = sc.nextInt() * 50;
        int oneHundredYenCoin = sc.nextInt() * 100;
        int fiveHundredYenCoin = sc.nextInt() * 500;

        int total = oneYenCoin + fiveYenCoin + tenYenCoin + fiftyYenCoin + oneHundredYenCoin + fiveHundredYenCoin;

        System.out.println(1000 <= total ? 1 : 0);

    }

}

File: aoj/volume2/n0296/Main.cpp
#include <iostream>

using namespace std;

int main() {

    int oneYenCoin, fiveYenCoin, tenYenCoin, fiftyYenCoin, oneHundredYenCoin, fiveHundredYenCoin;
    cin >> oneYenCoin >> fiveYenCoin >> tenYenCoin >> fiftyYenCoin >> oneHundredYenCoin >> fiveHundredYenCoin;

    int total = oneYenCoin + (fiveYenCoin * 5) + (tenYenCoin * 10) + (fiftyYenCoin * 50) + (oneHundredYenCoin * 100) + (fiveHundredYenCoin * 500);

    if (total >= 1000) {
        cout << 1 << endl;
    } else {
        cout << 0 << endl;
    }

    return 0;

}
